web/admin/src/editors.js

```diff
@@ -22,13 +22,64 @@ export class TextEditor extends PureComponent {
       <label className="text-editor">
         <div className="text-editor__title">{title}</div>
         { this.isValid() ? null : <div className="text-editor__error">{validationMessage}</div> }
-        <input
-          className="text-editor__input"
+        <div className="text-editor__inputBox">
+          <input
+            className="text-editor__input"
+            type="text"
+            placeholder={placeholder}
+            value={this.state.value}
+            onChange={this.onChange}
+            onBlur={this.onBlur}
+          />
+          {this.isTitle()}
+        </div>
+      </label>
+    )
+  }
+
+  isTitle = () => {
+    console.log(this.props.isTitle)
+    if (this.props.isTitle && this.state.value) {
+      return (
+        <p className="text-editor__counter">{150 - this.state.value.length} </p>
+      )
+    }
+  }
+
+  onChange = e => this.setState({value: e.target.value})
+  
+  onBlur = () => this.isValid() && this.props.onUpdate(this.props.prop, this.state.value)
+
+  isValid = () => {
+    const {regex} = this.props
+    const {value} = this.state
+    return !regex || value == null || value.match(regex)
+  }
+}
+
+export class MultiLineEditor extends PureComponent {
```

Comment:
Is this identical to the TextEditor, other than being a textarea rather than a single-line text input?  This would be a good candidate for refactoring the common pieces, which we can look at together.

Reply:
Correct